semeio/workflows/localisation/local_script_lib.py

```diff
@@ -788,19 +823,52 @@ def get_dx_dy(self, data_index):
         ) / self.perp_range
         return dx, dy
 
+    def norm_dist_square(self, data_index):
+        dx, dy = self.get_dx_dy(data_index)
+        d2 = dx * dx + dy * dy
+        return d2
+
 
 class GaussianDecay(Decay):
     def __call__(self, data_index):
-        dx, dy = super().get_dx_dy(data_index)
-        exp_arg = -3.0 * (dx * dx + dy * dy)
-        return math.exp(exp_arg)
+        d2 = super().norm_dist_square(data_index)
+        if self.use_one_in_range:
+            d = math.sqrt(d2)
+            if d <= 1.0:
+                return 1.0
+            if self.use_cutoff and d > self.normalised_tapering_range:
+                return 0.0
+
+            distance_from_inner_ellipse = (d - 1) / (self.normalised_tapering_range - 1)
+            exp_arg = -3 * distance_from_inner_ellipse * distance_from_inner_ellipse
+            return math.exp(exp_arg)
+
+        else:
+            if self.use_cutoff and d2 > 1.0:
+                return 0.0
+            exp_arg = -3.0 * d2
+            return math.exp(exp_arg)
```

Comment:
The functionality here looks good to me, but could it be an idea to introduce it as separate classes instead? For example:
```
class CutoffExpontentialDecay(GaussianDecay)
```
my main reason for suggesting that is that this has a lot of conditionals, which complicates things quite a bit for the user, for example if `use_one_in_range` if `False`, `normalised_tapering_range` is not used. By splitting this in a few different data models we can make it clearer which options do what.

Reply:
Rewritten using one class per method.